### PR TITLE
Added test script for np.nonzero()

### PR DESCRIPTION
**Pull request by tejalbarnwal**
added following script at `ulab/tests/2d/numpy/nonzero.py`
```python
try:
   from ulab import numpy as np
except:
   import numpy as np

array = np.array(range(16)).reshape((4,4))
boolean_array = array < 3

# np.nonzero() returns a tuple of corresponding row and column 
# indices array for each element being true in boolean array
row_indices, column_indices = np.nonzero(boolean_array)

print("Given Array:\n", array)
print("\nBoolean Array:\n", boolean_array)
print("\nFollowing indices have non-zero values:")
for i in range(len(row_indices)):
	print("row", row_indices[i], "and column", column_indices[i])
```
The expected output file is located at `ulab/tests/2d/numpy/nonzero.py.exp`
Expected output:
```
Given Array:
 array([[0.0, 1.0, 2.0, 3.0],
       [4.0, 5.0, 6.0, 7.0],
       [8.0, 9.0, 10.0, 11.0],
       [12.0, 13.0, 14.0, 15.0]], dtype=float64)

Boolean Array:
 array([[True, True, True, False],
       [False, False, False, False],
       [False, False, False, False],
       [False, False, False, False]], dtype=bool)

Following indices have non-zero values:
row 0 and column 0
row 0 and column 1
row 0 and column 2
```

### CODE REVIEW

**Comment by v923z** [comment]
`nonzero` works with arrays of arbitrary `dtype`, so e.g., this is also valid:

```python
>>> a = np.array(range(9), dtype=np.uint8).reshape((3,3)) - 4
>>> print(np.nonzero(a))
```
You can simply loop through the types like in https://github.com/v923z/micropython-ulab/blob/master/tests/2d/numpy/arange.py. Would you mind adding this, so that the test would be complete?

**Comment by tejalbarnwal** [comment]
sure, I will update with this

**Comment by v923z** [comment]
> sure, I will update with this

OK, thanks! Let me know when are ready. The rest of the PR is done, so as soon as you can update your test script, we could merge this into master.

File: tests/2d/numpy/nonzero.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+try:
+   from ulab import numpy as np
+except:
+   import numpy as np
+
+array = np.array(range(16)).reshape((4,4))
+boolean_array = array < 3
+
+# np.nonzero() returns a tuple of corresponding row and column
```
Comment (v923z) [inline tests/2d/numpy/nonzero.py:9]: Don't leave trivial comments in the test script! These are not even for humans, and there is the documentation, anyway.

File: tests/2d/numpy/nonzero.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+boolean_array = array < 3
+
+# np.nonzero() returns a tuple of corresponding row and column 
+# indices array for each element being true in boolean array
+row_indices, column_indices = np.nonzero(boolean_array)
+
+print("Given Array:\n", array)
+print("\nBoolean Array:\n", boolean_array)
+print("\nFollowing indices have non-zero values:")
+for i in range(len(row_indices)):
```
Comment (v923z) [inline tests/2d/numpy/nonzero.py:16]: You really don't need to loop through rows and columns. This is not a demonstration script, so you can just print out the results as in https://numpy.org/doc/stable/reference/generated/numpy.nonzero.html.

File: tests/2d/numpy/nonzero.py (hunk 1 above)
Comment (tejalbarnwal) [reply tests/2d/numpy/nonzero.py]: Hey, I removed all the comments

File: tests/2d/numpy/nonzero.py (hunk 2 above)
Comment (tejalbarnwal) [reply tests/2d/numpy/nonzero.py]: I removed this too

**Comment by tejalbarnwal** [comment]
I have updated the script to loop through different data types

File: tests/2d/numpy/nonzero.py (hunk 2 above)
Comment (v923z) [reply tests/2d/numpy/nonzero.py]: Thanks!

**Comment by v923z** [comment]
Fantastic! Many thanks for sticking with it till the very end! I hope you enjoyed the ride, and I am looking forward to your future contributions.

**Comment by tejalbarnwal** [comment]
Yes, sure

**Pull request merged**